### PR TITLE
Add EIP: Increase calldata cost

### PR DESCRIPTION
**Pull request by nerolation**
This EIP aims to reduce the maximum possible blocksize without impacting regular users.
This is achieved through a conditional expression that is used to determine the gas used.
Thereby, users that consume more than a certain threshold EVM gas per calldata byte will still have 16 gas calldata while DA users pay 68 gas per calldata byte.

### CODE REVIEW

File: EIPS/eip-7621.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+---
+eip: 7621
```
Comment (abcoathup) [inline EIPS/eip-7621.md:2]:
```suggestion
eip: 7623
```
Assigning next sequential EIP/ERC/RIP number.  EIP/ERC/RIP numbers are issued from a single set of numbers.  7621 was already issued to an ERC yesterday.
Numbering changed to sequential from 7500 and is no longer the PR number.  

Please also update the filename.

File: EIPS/eip-7621.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+title: Increase calldata cost
+description: Increase calldata cost to decrease the maximum block size
+author: Toni Wahrstätter (@nerolation), Vitalik Buterin (@vbuterin)
+discussions-to: https://ethereum-magicians.org/t/tbd/1234
```
Comment (abcoathup) [inline EIPS/eip-7621.md:6]: FYI: You can create a discussions topic on Eth Magicians before a number is issued.

Comment (nerolation) [reply EIPS/eip-7621.md]: Thanks - created one.

File: EIPS/eip-7621.md (hunk 1 above)
Comment (nerolation) [reply EIPS/eip-7621.md]: Updated

File: EIPS/eip-7623.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+This transition demands a reevaluation of calldata pricing, especially with regards to mitigating the inefficiency between the average block size in bytes and the maximum one possible.
+By increasing the gas cost for nonzero calldata bytes for transactions that are mainly using Ethereum for Data Availability (DA), the proposal aims to balance the need for block space with the necessity of reducing the maximum block size to make room for adding more blobs. The move from using calldata for DA to blobs further strengthens the multidimensional fee market by incentivizing blob space.
+
+Increasing calldata cost to 68 gas for transactions that do not spend more gas on EVM operations than a certain threshold, significantly reduces the maximum possible block size by lowering the number of huge, pure-DA transactions that fit into one block.
+However, the calldata price is kept at 16 gas per nonzero byte for those transactions that also spend enough on EVM computation, such that nothing changes for those users. Consequently, this approach safeguards users who depend on extensive calldata usage alongside EVM computations, ensuring they are not adversely affected by the new pricing model.
```
Comment (gaudren) [inline EIPS/eip-7623.md:30]: These paragraphs should be in the Rationale section since they explain the choice of gas cost.

Comment (nerolation) [reply EIPS/eip-7623.md]: Fixed

**Review by gaudren — APPROVED** [review]

**Review by SamWilsn — APPROVED** [review]

**Pull request merged**